### PR TITLE
feat!: add binding args

### PR DESCRIPTION
**Pull request by lucas-bremond**
Also removes deprecated overloads.

### CODE REVIEW

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Transformation.cpp [hunk 1]
```diff
@@ -11,19 +11,22 @@
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-inline void                     OpenSpaceToolkitMathematicsPy_Geometry_2D_Transformation   (     pybind11::module&  aModule                                     )
+inline void                     OpenSpaceToolkitMathematicsPy_Geometry_2D_Transformation ( pybind11::module&        aModule                                     )
 {
 
     using namespace pybind11 ;
 
+    using ostk::core::types::Unique ;
```
Comment (vishwa2710) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Transformation.cpp:19]: Unused?

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Transformation.cpp [hunk 2]
```diff
@@ -11,19 +11,22 @@
...
+
     using ostk::math::obj::Vector2d ;
     using ostk::math::obj::Matrix3d ;
-    using ostk::math::geom::d2::objects::Point ;
+    using ostk::math::geom::d2::Object ;
```
Comment (vishwa2710) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Transformation.cpp:23]: Unused as well

**Review by vishwa2710 — APPROVED** [review]
This is awesome, thanks for adding the args the bindings!

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/3D/Transformation.cpp [hunk 3]
```diff
@@ -39,24 +40,23 @@ inline void                     OpenSpaceToolkitMathematicsPy_Geometry_3D_Transf
...
+        .def("apply_to", overload_cast<const Vector3d&>(&Transformation::applyTo, const_), arg("vector"))
 
         .def_static("undefined", &Transformation::Undefined)
         .def_static("identity", &Transformation::Identity)
-        .def_static("translation", &Transformation::Translation)
-        // .def_static("rotation", &Transformation::Rotation)
-        // .def("rotation", +[] (const RotationVector& aRotationVector) -> Transformation { return Transformation::Rotation(aRotationVector) ; } )
-        .def("rotation", +[] (const RotationMatrix& aRotationMatrix) -> Transformation { return Transformation::Rotation(aRotationMatrix) ; } )
-        .def_static("rotation_around", &Transformation::RotationAround)
+        .def_static("translation", &Transformation::Translation, arg("transaction_vector"))
```
Comment (robinpdm) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/3D/Transformation.cpp:48]:
```suggestion
        .def_static("translation", &Transformation::Translation, arg("translation_vector"))
```

**Review by robinpdm — APPROVED** [review]

**Pull request merged**